src/components/datasets/DatasetOverview.js

```diff
@@ -4,14 +4,12 @@ import PropTypes from "prop-types";
 import { Col, Divider, Row, Spin, Statistic, Typography } from "antd";
 
 import { EM_DASH } from "@/constants";
-import { useDatasetDataTypes } from "@/modules/datasets/hooks";
+import { useDatasetDataTypesByID } from "@/modules/datasets/hooks";
 import { datasetPropTypesShape, projectPropTypesShape } from "@/propTypes";
 
 const DatasetOverview = ({ isPrivate, project, dataset }) => {
-  const datasetsDataTypes = useDatasetDataTypes();
-  const datasetDataTypes = datasetsDataTypes[dataset.identifier];
+  const datasetDataTypes = useDatasetDataTypesByID(dataset.identifier);
   const isFetchingDataset = datasetDataTypes?.isFetching;
-
   // Count data types which actually have data in them for showing in the overview
```

Comment:
we could use `isFetching` here from `datasetDataTypes` as another loading condition for the data types statistic

Reply:
I added `hasAttempted` in this store to only show the dash when waiting on the first fetch.

Reply:
nice!